### ISSUE TITLE
Ghidra fails to demangle some symbols

### ISSUE THREAD

**Issue by Martmists-GH** [NationalSecurityAgency/ghidra#5456]
**Describe the bug**
Many symbols are not demangled correctly by Ghidra.

**To Reproduce**
1. Create a symbol with one of the names below
2. Run the demangler
3. The symbol remains unchanged

**Expected behavior**
The symbol gets converted to a proper function as most others do.

**Attachments**
List of symbols not parsed correctly:
[failed_symbols.txt](https://github.com/NationalSecurityAgency/ghidra/files/11750141/failed_symbols.txt)

**Environment (please complete the following information):**
 - OS: Arch Linux
 - Java Version: OpenJDK 17
 - Ghidra Version: 10.3
 - Ghidra Origin: `extra/ghidra` from pacman

**Comment by Martmists-GH** [comment]
> Do you know what is applying the symbols to your target binary and in what order they are being created?

Yes, I have a script that applies symbols from a database, and explicitly marks them as primary. Afterwards, I run the demangler on the entire file. Some symbols were still present in the file, had already been demangled, the custom script re-sets the mangled as primary, demangler fails to demangle.

> Ghidra does not preserve the const-ness of types and it will also convert pass-by-reference to pointer for function signatures... they are currently treated the same internally.
Note that the Demangler internally identifies the const and reference notation, but when creating the function signature, it does not use this information.

Ghidra not preserving them is not a huge concern; The main issue is that it remains `_ZNK4sead10XmlElement18findAttributeValueERKNS_14SafeStringBaseIcEE` rather than `sead::XmlElement::findAttributeValue(sead::XmlElement* this,sead::SafeStringBase<char>* param_1)` which is what I'd expect.

**Comment by ghizard** [comment]
We believe that the label is being correctly demangled, but the program is not being updated with the result.  This is most likely due to the function or the signature not having SourceType.DEFAULT.

You can verify this if you are able to debug our source code with Eclipse.

Set breakpoint inside of AbstractDemanglerAnalyzer.doAdded().
Walk over/into functions to determine if something is preventing the Demangler from running.

Set breakpoitn inside of DemangledFunction.applyTo().
Walk over/into functions to determine what is preventing the individual symbol or function signature from being applied.

To test... set your label on the function.
Then from the Ghidra GUI... select the function signature in the Code Browser, then select Analysis -> One Shot -> Demangler GNU

**Comment by ghizard** [comment]
@Martmists-GH Are you able to do a build and run from source to test?  If not, you might try running your script prior to other analysis.

Keep in mind that there are four SourceType priority values: DEFAULT, ANALYSIS, IMPORT, and USER_DEFINED.  If your script is not setting labels with a higher SourceType priority than what is there, then your labels will not set.  If your script is the first thing to run, it should at least not run into issues of other analyzers setting a SourceType that you cannot overcome.  It is possible that we might need to change something so that your script can achieve this prerequisite at the time you are trying to run the script.

**Comment by ghizard** [comment]
@Martmists-GH 
Does this problem still persist for you?
Have you tried running only your script and then the Demangler?
When running your script after other analysis, but before the Demangler, have you checked the SourceType on the function and function signature?
